<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Run differential rhythmicity analysis — compareRhythms • compareRhythms</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Run differential rhythmicity analysis — compareRhythms"><meta name="description" content="The differential rhythmicity analysis is run with a call to this function. To
execute this function, the three necessary ingredients are the timeseries
data, the experimental design and parameters to choose and tune the method."><meta property="og:description" content="The differential rhythmicity analysis is run with a call to this function. To
execute this function, the three necessary ingredients are the timeseries
data, the experimental design and parameters to choose and tune the method."><meta property="og:image" content="https://bharathananth.github.io/compareRhythms/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">compareRhythms</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/UserGuide.html">User guide</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/bharathananth/compareRhythms/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Run differential rhythmicity analysis</h1>
      <small class="dont-index">Source: <a href="https://github.com/bharathananth/compareRhythms/blob/master/R/compareRhythms.R" class="external-link"><code>R/compareRhythms.R</code></a></small>
      <div class="d-none name"><code>compareRhythms.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>The differential rhythmicity analysis is run with a call to this function. To
execute this function, the three necessary ingredients are the timeseries
data, the experimental design and parameters to choose and tune the method.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">compareRhythms</span><span class="op">(</span></span>
<span>  <span class="va">data</span>,</span>
<span>  <span class="va">exp_design</span>,</span>
<span>  lengths <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"mod_sel"</span>,</span>
<span>  period <span class="op">=</span> <span class="fl">24</span>,</span>
<span>  rhythm_fdr <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  compare_fdr <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  amp_cutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  criterion <span class="op">=</span> <span class="st">"bic"</span>,</span>
<span>  schwarz_wt_cutoff <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>  just_classify <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  robust <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  outliers <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  longitudinal <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  just_rhythms <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-data">data<a class="anchor" aria-label="anchor" href="#arg-data"></a></dt>
<dd><p>A matrix of log2 expression values (if microarray), expression
counts (RNA-seq) or normalized data (see Details).</p></dd>


<dt id="arg-exp-design">exp_design<a class="anchor" aria-label="anchor" href="#arg-exp-design"></a></dt>
<dd><p>A data.frame of the experimental design with at least two
columns: "time" and "group".</p></dd>


<dt id="arg-lengths">lengths<a class="anchor" aria-label="anchor" href="#arg-lengths"></a></dt>
<dd><p>A data.frame of average transcript lengths. Only used with
methods "deseq" and "edgeR".</p></dd>


<dt id="arg-method">method<a class="anchor" aria-label="anchor" href="#arg-method"></a></dt>
<dd><p>The method of analysis. It should be one of "mod_sel" for model
selection, "dodr" for analysis using <a href="https://rdrr.io/pkg/DODR/man/dodr.html" class="external-link">DODR::dodr</a>, "limma" for
linear-modeling approach based on <span class="pkg">limma</span>, "voom" for linear-modeling
approach for RNA-Seq using <a href="https://rdrr.io/pkg/limma/man/voom.html" class="external-link">limma::voom</a>, "deseq2" for RNA-seq analysis
using <span class="pkg">DESeq2</span>, "edger" for RNA-seq analysis using <span class="pkg">edgeR</span>, and
"cosinor" for simple cosinor-based analysis both for independent samples
and repeated samples.</p></dd>


<dt id="arg-period">period<a class="anchor" aria-label="anchor" href="#arg-period"></a></dt>
<dd><p>The period of rhythm being tested (default = 24)</p></dd>


<dt id="arg-rhythm-fdr">rhythm_fdr<a class="anchor" aria-label="anchor" href="#arg-rhythm-fdr"></a></dt>
<dd><p>The false discovery cutoff for finding rhythmic time series
(default = 0.05)</p></dd>


<dt id="arg-compare-fdr">compare_fdr<a class="anchor" aria-label="anchor" href="#arg-compare-fdr"></a></dt>
<dd><p>The false discovery cutoff for the comparison of rhythms
(default = 0.05)</p></dd>


<dt id="arg-amp-cutoff">amp_cutoff<a class="anchor" aria-label="anchor" href="#arg-amp-cutoff"></a></dt>
<dd><p>The minimum peak-to-trough amplitude in log2 scale
considered biologically relevant (default = 0.5)</p></dd>


<dt id="arg-criterion">criterion<a class="anchor" aria-label="anchor" href="#arg-criterion"></a></dt>
<dd><p>The criterion used for model selection. These can be "aic"
or "bic" (default = "bic"). Only used for method = "mod_sel".</p></dd>


<dt id="arg-schwarz-wt-cutoff">schwarz_wt_cutoff<a class="anchor" aria-label="anchor" href="#arg-schwarz-wt-cutoff"></a></dt>
<dd><p>The conditional probability that the best model is
the true model. Genes with a conditional probability smaller than this
cutoff are deemed unclassifiable. This is only used for method = "mod_sel".
(default = 0.4)</p></dd>


<dt id="arg-just-classify">just_classify<a class="anchor" aria-label="anchor" href="#arg-just-classify"></a></dt>
<dd><p>Boolean specifying whether genes must only be classified
(TRUE) or if the amplitude and phases of fits should also be returned
(FALSE)</p></dd>


<dt id="arg-robust">robust<a class="anchor" aria-label="anchor" href="#arg-robust"></a></dt>
<dd><p>Boolean to turn on robust computation of statistics in
different methods (default = TRUE).</p></dd>


<dt id="arg-outliers">outliers<a class="anchor" aria-label="anchor" href="#arg-outliers"></a></dt>
<dd><p>Boolean specifying if weights must be computed for each
sample to account for outliers. Only used by method = "voom".</p></dd>


<dt id="arg-longitudinal">longitudinal<a class="anchor" aria-label="anchor" href="#arg-longitudinal"></a></dt>
<dd><p>Boolean specifying if repeated samples from one
experimental unit. Only used by method = "cosinor".</p></dd>


<dt id="arg-just-rhythms">just_rhythms<a class="anchor" aria-label="anchor" href="#arg-just-rhythms"></a></dt>
<dd><p>Boolean specifying whether only rhythm analysis (True,
default) or differential expression/magnitude analysis should also be
performed.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A <em>data.frame</em> with the names of the differentially rhythmic
features, the category it is classified under and optionally the rhythm
parameters of the features in each group. The differential rhythmicity
categories are <strong>gain</strong> of, <strong>loss</strong> of, <strong>change</strong> of, or <strong>same</strong> rhythms
(with respect to the reference/control group).</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Bharath Ananthasubramaniam.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

