<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Analyzing rhythmic data with compareRhythms • compareRhythms</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Analyzing rhythmic data with compareRhythms">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">compareRhythms</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item"><a class="nav-link" href="../articles/UserGuide.html">User guide</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/bharathananth/compareRhythms/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Analyzing rhythmic data with compareRhythms</h1>
                        <h4 data-toc-skip class="author">Bharath
Ananthasubramaniam</h4>
            
            <h4 data-toc-skip class="date">02 Jun 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/bharathananth/compareRhythms/blob/master/vignettes/UserGuide.Rmd" class="external-link"><code>vignettes/UserGuide.Rmd</code></a></small>
      <div class="d-none name"><code>UserGuide.Rmd</code></div>
    </div>

    
    
<p>This package is designed to find features with altered circadian
rhythm parameters (<em>amplitude</em> and <em>phase</em>) between the
control and experimental groups. In this vignette, we will walk through
two examples that show the basic application of the
<code>compareRhythms</code> package to microarray and RNA-seq data.
Nevertheless, any rhythmic data can be analyzed using this package.</p>
<div class="section level3">
<h3 id="usage-summary">Usage summary<a class="anchor" aria-label="anchor" href="#usage-summary"></a>
</h3>
<p>The analysis is run using a single function
<code><a href="../reference/compareRhythms.html">compareRhythms()</a></code>. To execute this function, the three
necessary ingredients are the timeseries data, the experimental design
and parameters to choose and tune the method. The output of the function
is a <em>data.frame</em> with the IDs of the differentially rhythmic
features, the category they are classified under and optionally the
rhythm parameters of the features in the two groups. The differential
rhythmicity categories are <strong>gain</strong> of,
<strong>loss</strong> of, <strong>change</strong> of, or
<strong>same</strong> rhythms (with respect to the reference/control
group).</p>
</div>
<div class="section level3">
<h3 id="time-series-data-and-experimental-design">Time series data and experimental design<a class="anchor" aria-label="anchor" href="#time-series-data-and-experimental-design"></a>
</h3>
<p>Only two inputs are mandatory to run the workflows in this
package.</p>
<ol style="list-style-type: decimal">
<li><p>A single (numeric) <em>matrix</em> combining both timeseries
datasets. The rows of this matrix are the features and the columns are
the different samples. The <em>rownames</em> of this data matrix provide
the <code>id</code> list for the features.</p></li>
<li>
<p>A <em>data.frame</em> specifying the experimental design (details
regarding each sample) in order to interpret the data matrix.</p>
<ol style="list-style-type: lower-roman">
<li>There must be one row in this data.frame describing each sample
(column) in the data matrix.</li>
<li>This data.frame must contain a numeric column named
<code>time</code> specifying the time associated with the sample and a
<em>factor</em> column named <code>group</code> specifying whether a
sample belongs to the control or the experimental group. This
<code>group</code> variable must have only two levels with one chosen as
the reference with respect to which the results are presented.</li>
<li>Optionally, a column named <code>batch</code> can be used to specify
a categorical (factor) covariate representing an independent or
confounding variable (Note: covariates cannot be included in all
methods. More on that below). Currently, there is no capability to
include continuous covariates in the analysis.</li>
</ol>
</li>
</ol>
</div>
<div class="section level3">
<h3 id="choice-of-method">Choice of method<a class="anchor" aria-label="anchor" href="#choice-of-method"></a>
</h3>
<p>The package currently offers a choice of 6 different methods. We
describe below the tuning parameters available for each method.</p>
<p>There are three parameters that are common to all methods.</p>
<ul>
<li><p><code>period</code> : This positive number is the period of the
rhythms, whose amplitude and phase must be compared between the two
timeseries datasets. Defaults to 24.</p></li>
<li><p><code>amp_cutoff</code> : Only features with a peak-to-trough
amplitude greater than this positive number in at least one group are
included in the differential rhythmicity results.</p></li>
<li><p><code>just_classify</code> : This boolean flag specifies if the
amplitude and phase estimates of each differential rhythmic feature in
the two groups (<code>just_classify = FALSE</code>) must be returned in
addition to the list of <code>id</code>s and their classification into
the differential rhythmicity categories.</p></li>
</ul>
<p>The different approaches can be divided into model selection and the
rest that are all implementations of hypothesis testing:</p>
<ol style="list-style-type: decimal">
<li>
<dl>
<dt>
<strong>Model selection</strong> (<code>method = "mod_sel"</code>
(default))</dt>
<dd>
The different categories of differentially rhythmic features are
represented by linear regression models that are fit to the data <span class="citation">(Atger et al. 2015)</span>. The best model/category is
selected by an information theoretic <code>criterion</code>. The quality
of fit of the best category to the data is fine tuned using
<code>schwarz_wt_cutoff</code>.
<ul>
<li>
<code>criterion</code> is used to select the desired information
criterion to pick the best model/category. “bic” (default) selects
Bayesian Information Criterion (BIC) and “aic” selects Akaike
Information Criterion (AIC). BIC penalizes model size more than AIC and
hence favors smaller models.</li>
<li>
<code>schwarz_wt_cutoff</code> (default = 0.6) is a probability
threshold for the weight of the best model/category. The weight of the
best model/category (called Schwarz weight with BIC and Akaike weight
with AIC) is the probability that the chosen category is the best
category given the data and the other models/categories. Higher this
number (between 0 and 1), more certain is the classification. But models
that do not reach this threshold are left unclassified.</li>
</ul>
</dd>
</dl>
<p><u>Use cases</u>: This method can be used on any normalized data.
Technically, the assumption that the noise/errors is/are normal at each
sample and independent needs to be acceptable.</p>
</li>
<li>
<dl>
<dt>
<strong>DODR</strong> (<code>method = "dodr"</code>)</dt>
<dd>
Rhythmic features in either group are first identified using <a href="https://doi.org/doi:10.18129/B9.bioc.rain" class="external-link">rain</a> followed by
filtering by rhythm amplitude (<code>amp_cutoff</code>). The resulting
subset of features are processed using <a href="https://cran.r-project.org/package=DODR" class="external-link">DODR</a> <span class="citation">(Thaben and Westermark 2016)</span>, an R package to
find the differentially rhythmic features. The features are classified
after the fact into the 4 categories listed above.
<ul>
<li>
<code>rhythm_fdr</code> is the threshold for pre-filtering rhythmic
features in either group based on the multiple testing corrected p-value
from rain.</li>
<li>
<code>compare_fdr</code> is the threshold for selecting
differentially rhythmic features using the multiple testing corrected
p-values from DODR.</li>
</ul>
</dd>
</dl>
<p><u>Use cases</u>: This method can be used on any normalized
data.</p>
</li>
<li>
<dl>
<dt>
<strong>limma</strong> (<code>method = "limma"</code>)</dt>
<dd>
This is an implementation of <strong>DODR</strong> using the linear
modeling approach for microarrays in <a href="https://doi.org/doi:10.18129/B9.bioc.limma" class="external-link">limma</a> <span class="citation">(Ritchie et al. 2015)</span>. The analysis follows the
same steps with the difference that the filtering for rhythmic features
in either group can be accomplished in a single test and differential
rhythmicity test is also achieved in the same framework by comparing
rhythm parameters (amplitude and phase).
<ul>
<li>
<code>rhythm_fdr</code> is the threshold for pre-filtering rhythmic
features in either group based on the multiple testing corrected
p-value.</li>
<li>
<code>compare_fdr</code> is the threshold for selecting
differentially rhythmic features using the multiple testing corrected
p-values</li>
<li>
<code>robust</code> is a boolean to make the noise estimates for
individual features in limma robust against outliers (see
<code>eBayes</code> in <a href="https://doi.org/doi:10.18129/B9.bioc.limma" class="external-link">limma</a> for
details).</li>
</ul>
</dd>
</dl>
<p><u>Use cases</u>: This method is to be used only on log normalized
microarray data (see limma for details).</p>
</li>
<li>
<dl>
<dt>
<strong>voom</strong> (<code>method = "voom"</code>)</dt>
<dd>
This is a variation of <strong>limma</strong> to also process RNA-seq
data with the same pipeline. Count data are preprocessed using
<code>voom</code> and then analyzed using the <strong>limma</strong>
method above.
<ul>
<li>
<code>rhythm_fdr</code> (see <strong>limma</strong>)</li>
<li>
<code>compare_fdr</code> (see <strong>limma</strong>)</li>
<li>
<code>robust</code> (see <strong>limma</strong>)</li>
<li>
<code>outliers</code> is a boolean to downweight outlier
<em>samples</em> in the analysis. (see
<code>voomWithQualityWeights</code> in <a href="https://doi.org/doi:10.18129/B9.bioc.limma" class="external-link">limma</a> for
details)</li>
</ul>
</dd>
</dl>
<p><u>Use cases</u>: This method is to be used with count data from an
RNA-seq experiment. Count data from aligment (STAR, TopHat2) followed by
quantification (htseq-count, summarizeOverlaps, featureCounts) can be
directly used. If <a href="https://doi.org/doi:10.18129/B9.bioc.tximport" class="external-link">tximport</a> is
used to import data, then use counts after setting
<code>countsFromAbundance = "lengthScaledTPM" or "scaledTPM"</code> in
the <code>tximport()</code> call.</p>
</li>
<li>
<dl>
<dt>
<strong>DESeq2</strong> (<code>method = "deseq2"</code>)</dt>
<dd>
This is the <strong>limma</strong> workflow adapted to process RNA-seq
data according to <a href="https://doi.org/10.18129/B9.bioc.DESeq2" class="external-link">DESeq2</a> <span class="citation">(Love, Huber, and Anders 2014)</span>.
<ul>
<li>
<code>rhythm_fdr</code> (see <strong>limma</strong>)</li>
<li>
<code>compare_fdr</code> (see <strong>limma</strong>)</li>
<li>
<code>length</code> is an optional <em>matrix</em> (with the same
size as data) containing the average transcript length of each gene in
each sample.</li>
</ul>
</dd>
</dl>
<p><u>Use cases</u>: This method is to be used with count data from an
RNA-seq experiment. Count data from aligment (STAR, TopHat2) followed by
quantification (htseq-count, summarizeOverlaps, featureCounts) can be
directly used. If <a href="https://doi.org/doi:10.18129/B9.bioc.tximport" class="external-link">tximport</a> is
used to import data, then use <code>counts</code> and
<code>length</code> obtained from the <code>tximport()</code> call with
<code>countsFromAbundance = "no"</code>.</p>
</li>
<li>
<dl>
<dt>
<strong>edgeR</strong> (<code>method = "edger"</code>)</dt>
<dd>
This is the <strong>limma</strong> workflow adapted to process RNA-seq
data according to <a href="https://doi.org/10.18129/B9.bioc.edgeR" class="external-link">edgeR</a> <span class="citation">(Robinson, McCarthy, and Smyth 2010)</span>.
<ul>
<li>
<code>rhythm_fdr</code> (see <strong>limma</strong>)</li>
<li>
<code>compare_fdr</code> (see <strong>limma</strong>)</li>
<li>
<code>length</code> (see <strong>DESeq2</strong>)</li>
</ul>
</dd>
</dl>
<p><u>Use cases</u>: This method is to be used with count data from an
RNA-seq experiment. Count data from aligment (STAR, TopHat2) followed by
quantification (htseq-count, summarizeOverlaps, featureCounts) can be
directly used. If <a href="https://doi.org/doi:10.18129/B9.bioc.tximport" class="external-link">tximport</a> is
used to import data, then use <code>counts</code> and
<code>length</code> obtained from the <code>tximport()</code> call with
<code>countsFromAbundance = "no"</code>.</p>
</li>
<li>
<p><strong>Cosinor</strong> (<code>method = "cosinor"</code>): This
implements the simple classical <strong>cosinor</strong> analysis and
includes an option to deal with longitudinal data (which is common in
data from human studies).</p>
<ul>
<li>
<code>rhythm_fdr</code> (see <strong>limma</strong>)</li>
<li>
<code>compare_fdr</code> (see <strong>limma</strong>)</li>
<li>
<code>longitudinal</code> is a boolean to switch between
<em>independent</em> time samples (FALSE, default) or
<em>repeated-measures</em> samples (TRUE).</li>
</ul>
<p><u>Use cases</u>: This method can be used for any normalized dataset
of moderate size (few hundreds of features). This analysis assumes that
noise in the data is Gaussian and that there are no trends in the
measurements across time. The <code>longitudinal = TRUE</code> requires
the user to provide a factor column <code>ID</code> in the
<code>exp_design</code> with the identity of each experimental unit that
is repeatedly measured (Note: this analysis uses mixed-model framework
of <a href="https://cran.r-project.org/package=lme4" class="external-link">lme4</a>).</p>
</li>
</ol>
</div>
<div class="section level3">
<h3 id="example-1-microarray-data">Example 1: Microarray data<a class="anchor" aria-label="anchor" href="#example-1-microarray-data"></a>
</h3>
<p>We analyze first the microarray data on the changes in circadian
liver transcriptome under high fat diet (HFD) with respect to normal
chow (NC) <span class="citation">(Eckel-Mahan et al. 2013)</span>. This
data is provided with this package as a 23060x36 matrix called
<code>high_fat_diet_ma</code>. Liver transcripts were quantified every
4h for 24h (6 samples) and the 2 different conditions with 3 biological
replicates each (36=6x2x3). <a href="https://bioconductor.org/packages/devel/workflows/vignettes/maEndToEnd/inst/doc/MA-Workflow.html" class="external-link">maEndToEnd</a>
describes how to perform quality control and normalization for
microarrays.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/bharathananth/compareRhythms" class="external-link">compareRhythms</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">high_fat_diet_ma</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;                     NC_ZT0_1  NC_ZT0_2  NC_ZT0_3 HFD_ZT0_1 HFD_ZT0_2 HFD_ZT0_3</span></span>
<span><span class="co">#&gt; ENSMUSG00000000001  9.404793  9.328668  9.310453  9.335252  9.470418  9.511085</span></span>
<span><span class="co">#&gt; ENSMUSG00000000003  2.470968  2.358439  2.502844  2.530175  2.453542  2.562028</span></span>
<span><span class="co">#&gt; ENSMUSG00000000028  6.017191  5.769798  5.707153  6.015197  5.702113  5.751801</span></span>
<span><span class="co">#&gt; ENSMUSG00000000031  5.317120  5.010673  5.068763  5.107669  5.193487  4.935580</span></span>
<span><span class="co">#&gt; ENSMUSG00000000037  3.672698  3.670284  3.971859  3.556922  3.798398  3.641636</span></span>
<span><span class="co">#&gt; ENSMUSG00000000049 12.677798 12.739597 12.590512 12.699638 12.782454 12.613611</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">high_fat_diet_ma</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "NC_ZT0_1"   "NC_ZT0_2"   "NC_ZT0_3"   "HFD_ZT0_1"  "HFD_ZT0_2" </span></span>
<span><span class="co">#&gt;  [6] "HFD_ZT0_3"  "NC_ZT4_1"   "NC_ZT4_2"   "NC_ZT4_3"   "HFD_ZT4_1" </span></span>
<span><span class="co">#&gt; [11] "HFD_ZT4_2"  "HFD_ZT4_3"  "NC_ZT8_1"   "NC_ZT8_2"   "NC_ZT8_3"  </span></span>
<span><span class="co">#&gt; [16] "HFD_ZT8_1"  "HFD_ZT8_2"  "HFD_ZT8_3"  "NC_ZT12_1"  "NC_ZT12_2" </span></span>
<span><span class="co">#&gt; [21] "NC_ZT12_3"  "HFD_ZT12_1" "HFD_ZT12_2" "HFD_ZT12_3" "NC_ZT16_1" </span></span>
<span><span class="co">#&gt; [26] "NC_ZT16_2"  "NC_ZT16_3"  "HFD_ZT16_1" "HFD_ZT16_2" "HFD_ZT16_3"</span></span>
<span><span class="co">#&gt; [31] "NC_ZT20_1"  "NC_ZT20_2"  "NC_ZT20_3"  "HFD_ZT20_1" "HFD_ZT20_2"</span></span>
<span><span class="co">#&gt; [36] "HFD_ZT20_3"</span></span></code></pre></div>
<p>We will first construct the data.frame of the experimental design. In
this simple example, the required experimental design information is
encoded in the column names, which we extract.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exp_design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_split.html" class="external-link">str_split</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">high_fat_diet_ma</span><span class="op">)</span>, <span class="st">"_"</span>, simplify <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>     <span class="co"># split the names by _</span></span>
<span><span class="va">exp_design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">exp_design</span>, stringsAsFactors<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>    <span class="co"># convert from matrix to data.frame</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">exp_design</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"group"</span>, <span class="st">"time"</span>, <span class="st">"rep"</span><span class="op">)</span>    <span class="co"># name the columns</span></span>
<span><span class="va">exp_design</span><span class="op">$</span><span class="va">time</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"ZT"</span>, <span class="st">""</span>, <span class="va">exp_design</span><span class="op">$</span><span class="va">time</span><span class="op">)</span><span class="op">)</span> <span class="co"># remove ZT prefix and make numeric</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">exp_design</span><span class="op">)</span></span>
<span><span class="co">#&gt;   group time rep</span></span>
<span><span class="co">#&gt; 1    NC    0   1</span></span>
<span><span class="co">#&gt; 2    NC    0   2</span></span>
<span><span class="co">#&gt; 3    NC    0   3</span></span>
<span><span class="co">#&gt; 4   HFD    0   1</span></span>
<span><span class="co">#&gt; 5   HFD    0   2</span></span>
<span><span class="co">#&gt; 6   HFD    0   3</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">exp_design</span><span class="op">)</span>   <span class="co"># view data type of each column</span></span>
<span><span class="co">#&gt; 'data.frame':    36 obs. of  3 variables:</span></span>
<span><span class="co">#&gt;  $ group: Factor w/ 2 levels "HFD","NC": 2 2 2 1 1 1 2 2 2 1 ...</span></span>
<span><span class="co">#&gt;  $ time : num  0 0 0 0 0 0 4 4 4 4 ...</span></span>
<span><span class="co">#&gt;  $ rep  : Factor w/ 3 levels "1","2","3": 1 2 3 1 2 3 1 2 3 1 ...</span></span></code></pre></div>
<p>So <code>exp_design</code> has the two required columns
<em>group</em> and <em>time</em>. Furthermore, <em>group</em> is (as
required) a factor with only 2 levels and <em>time</em> is numeric. We
can also check what the reference group is.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">exp_design</span><span class="op">$</span><span class="va">group</span><span class="op">)</span> <span class="co"># the first level is the reference group</span></span>
<span><span class="co">#&gt; [1] "HFD" "NC"</span></span>
<span><span class="va">exp_design</span><span class="op">$</span><span class="va">group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/relevel.html" class="external-link">relevel</a></span><span class="op">(</span><span class="va">exp_design</span><span class="op">$</span><span class="va">group</span>, <span class="st">"NC"</span><span class="op">)</span> <span class="co"># choose NC as the correct reference group</span></span></code></pre></div>
<p>It is also useful at this point to check whether there are any
outlier samples (columns) using PCA.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pca_ma</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/prcomp.html" class="external-link">prcomp</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">high_fat_diet_ma</span><span class="op">)</span>, scale. <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">varExp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">pca_ma</span><span class="op">$</span><span class="va">sdev</span><span class="op">^</span><span class="fl">2</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">pca_ma</span><span class="op">$</span><span class="va">sdev</span><span class="op">^</span><span class="fl">2</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>PC1 <span class="op">=</span> <span class="va">pca_ma</span><span class="op">$</span><span class="va">x</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>, PC2 <span class="op">=</span> <span class="va">pca_ma</span><span class="op">$</span><span class="va">x</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span>,</span>
<span>                 diet <span class="op">=</span> <span class="va">exp_design</span><span class="op">$</span><span class="va">group</span>,</span>
<span>                 time <span class="op">=</span> <span class="va">exp_design</span><span class="op">$</span><span class="va">time</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">PC1</span>, <span class="va">PC2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">time</span>, color<span class="op">=</span><span class="va">diet</span><span class="op">)</span>, size<span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span>base_size<span class="op">=</span><span class="fl">10</span><span class="op">)</span> <span class="op">+</span>   <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_color_brewer</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Diet"</span>, palette<span class="op">=</span><span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"PC1, VarExp: "</span>, <span class="va">varExp</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="st">"%"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"PC2, VarExp: "</span>, <span class="va">varExp</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, <span class="st">"%"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>The first principal component (PC) seems to capture the time
variation and the second PC captures differences in diet. Therefore, the
main variations we are interested in are captured in top 2 PCs. There
appear to be no obvious outliers.</p>
<p>All <code>23060</code> genes (features) in the mouse transcriptome
(according to Ensembl) are included in the data. We want to keep only
the strongly enough expressed genes for differential rhythmicity
analysis.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">grp_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">exp_design</span><span class="op">$</span><span class="va">group</span><span class="op">)</span>   <span class="co"># extract the names of the two groups</span></span>
<span><span class="va">mean_g1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowMeans</a></span><span class="op">(</span><span class="va">high_fat_diet_ma</span><span class="op">[</span>, <span class="va">exp_design</span><span class="op">$</span><span class="va">group</span> <span class="op">==</span> <span class="va">grp_ids</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="co"># mean expression of group 1</span></span>
<span><span class="va">mean_g2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowMeans</a></span><span class="op">(</span><span class="va">high_fat_diet_ma</span><span class="op">[</span>, <span class="va">exp_design</span><span class="op">$</span><span class="va">group</span> <span class="op">==</span> <span class="va">grp_ids</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="co"># mean expression of group 2</span></span>
<span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>mean<span class="op">=</span><span class="va">mean_g1</span>, group<span class="op">=</span><span class="va">grp_ids</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>mean<span class="op">=</span><span class="va">mean_g2</span>, group<span class="op">=</span><span class="va">grp_ids</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>    <span class="co"># data.frame for plotting</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">stat_bin</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">mean</span>, fill<span class="op">=</span><span class="va">group</span><span class="op">)</span>, bins <span class="op">=</span> <span class="fl">100</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">group</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span>base_size<span class="op">=</span><span class="fl">10</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">5.5</span>, color<span class="op">=</span><span class="st">"grey50"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>strip.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"mean expression"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span>fill<span class="op">=</span><span class="st">"none"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"number of genes"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_brewer</a></span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Dark2"</span><span class="op">)</span></span></code></pre></div>
<p><img src="UserGuide_files/figure-html/histogram-expression-1.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">mean_g1</span> <span class="op">&gt;</span> <span class="fl">5.5</span><span class="op">)</span> <span class="op">|</span> <span class="op">(</span><span class="va">mean_g2</span> <span class="op">&gt;</span> <span class="fl">5.5</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">keep</span><span class="op">)</span>    <span class="co"># summary of how many genes will be kept after filtering</span></span>
<span><span class="co">#&gt; keep</span></span>
<span><span class="co">#&gt; FALSE  TRUE </span></span>
<span><span class="co">#&gt;  9898 13162</span></span>
<span><span class="va">expr_filtered</span> <span class="op">&lt;-</span> <span class="va">high_fat_diet_ma</span><span class="op">[</span><span class="va">keep</span>, <span class="op">]</span></span></code></pre></div>
<p>Before we run <code><a href="../reference/compareRhythms.html">compareRhythms()</a></code> on this data let us check
that * the number of columns in <code>expr_filtered</code> match number
of rows of <code>exp_design</code>. * <code>expr_filtered</code> is a
matrix and <code>exp_design</code> is a data.frame</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">exp_design</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">expr_filtered</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">exp_design</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "data.frame"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">expr_filtered</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "matrix" "array"</span></span></code></pre></div>
<p>We will analyze this microarray data using <em>model selection</em>
first. Log2 normalized data such as <code>high_fat_diet_ma</code> can be
directly used.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compareRhythms.html">compareRhythms</a></span><span class="op">(</span><span class="va">expr_filtered</span>, exp_design <span class="op">=</span> <span class="va">exp_design</span>, </span>
<span>                          period <span class="op">=</span> <span class="fl">24</span>, method <span class="op">=</span> <span class="st">"mod_sel"</span><span class="op">)</span> <span class="co"># run with default parameters for schwarz_wt_cutoff and criterion</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></span>
<span><span class="co">#&gt;                   id category</span></span>
<span><span class="co">#&gt; 1 ENSMUSG00000000001    arrhy</span></span>
<span><span class="co">#&gt; 2 ENSMUSG00000000028    arrhy</span></span>
<span><span class="co">#&gt; 3 ENSMUSG00000000049    arrhy</span></span>
<span><span class="co">#&gt; 4 ENSMUSG00000000056   change</span></span>
<span><span class="co">#&gt; 5 ENSMUSG00000000058    arrhy</span></span>
<span><span class="co">#&gt; 6 ENSMUSG00000000078    arrhy</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">category</span><span class="op">)</span>    <span class="co"># number of genes in the different categories</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  arrhy   loss   gain   same change </span></span>
<span><span class="co">#&gt;   8911    137     27    501    158</span></span></code></pre></div>
<p>For model selection alone, the number of arrhythmic features is
returned, since some features that have weights less than
<code>schwarz_wt_cutoff</code> are unclassified. Here,
<code>13162</code> - <code>9734</code>= <code>3428</code> genes are left
unclassified.</p>
<p>If the peak-to-trough amplitude and peak phase (acrophase) estimates
are desired,</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compareRhythms.html">compareRhythms</a></span><span class="op">(</span><span class="va">expr_filtered</span>, exp_design <span class="op">=</span> <span class="va">exp_design</span>, period <span class="op">=</span> <span class="fl">24</span>, </span>
<span>                          method <span class="op">=</span> <span class="st">"mod_sel"</span>, just_classify <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="co"># run with default parameters for schwarz_wt_cutoff and criterion</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></span>
<span><span class="co">#&gt;                   id category   NC_amp NC_phase   HFD_amp HFD_phase   weights</span></span>
<span><span class="co">#&gt; 1 ENSMUSG00000000001    arrhy 0.000000 0.000000 0.0000000  0.000000 0.7949295</span></span>
<span><span class="co">#&gt; 2 ENSMUSG00000000028    arrhy 0.000000 0.000000 0.0000000  0.000000 0.7822633</span></span>
<span><span class="co">#&gt; 3 ENSMUSG00000000049    arrhy 0.000000 0.000000 0.0000000  0.000000 0.6327680</span></span>
<span><span class="co">#&gt; 4 ENSMUSG00000000056   change 1.097253 2.373042 0.7517809  2.180985 0.6574918</span></span>
<span><span class="co">#&gt; 5 ENSMUSG00000000058    arrhy 0.000000 0.000000 0.0000000  0.000000 0.8867482</span></span>
<span><span class="co">#&gt; 6 ENSMUSG00000000078    arrhy 0.000000 0.000000 0.0000000  0.000000 0.6069374</span></span></code></pre></div>
<p>The amplitude is the units of the data (for this method) and phase is
in radians. To get the phase in h, multiply the phase estimates by
<code>period/(2*pi)</code>.</p>
<p>Next, we analyze the data using the <em>DODR</em> method. We only
show entire results (with <code>just_classify = FALSE</code>) for
simplicity.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compareRhythms.html">compareRhythms</a></span><span class="op">(</span><span class="va">expr_filtered</span>, exp_design <span class="op">=</span> <span class="va">exp_design</span>, period <span class="op">=</span> <span class="fl">24</span>, </span>
<span>                          method <span class="op">=</span> <span class="st">"dodr"</span>, just_classify <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="co"># run with default parameters</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></span>
<span><span class="co">#&gt;                   id category rhythmic_in_NC rhythmic_in_HFD diff_rhythmic</span></span>
<span><span class="co">#&gt; 1 ENSMUSG00000000056     same           TRUE            TRUE         FALSE</span></span>
<span><span class="co">#&gt; 2 ENSMUSG00000000555     same           TRUE            TRUE         FALSE</span></span>
<span><span class="co">#&gt; 3 ENSMUSG00000000567     same           TRUE            TRUE         FALSE</span></span>
<span><span class="co">#&gt; 4 ENSMUSG00000000708     same           TRUE           FALSE         FALSE</span></span>
<span><span class="co">#&gt; 5 ENSMUSG00000000876     same           TRUE            TRUE         FALSE</span></span>
<span><span class="co">#&gt; 6 ENSMUSG00000001018     same           TRUE           FALSE         FALSE</span></span>
<span><span class="co">#&gt;      NC_amp  NC_phase    HFD_amp HFD_phase adj_p_val_NC adj_p_val_HFD</span></span>
<span><span class="co">#&gt; 1 1.0972527 2.3730421 0.75178092  2.180985 1.088656e-05  9.814312e-06</span></span>
<span><span class="co">#&gt; 2 1.1161226 5.5799337 0.90190671  5.400475 2.045434e-05  2.755708e-05</span></span>
<span><span class="co">#&gt; 3 0.8647148 0.2533993 0.62547925  6.263871 4.117639e-04  1.503105e-03</span></span>
<span><span class="co">#&gt; 4 0.5127966 2.4419457 0.04522975  2.532181 2.210451e-03  9.421081e-01</span></span>
<span><span class="co">#&gt; 5 0.7570742 5.9485326 0.76704298  5.346225 3.682985e-05  8.200630e-05</span></span>
<span><span class="co">#&gt; 6 0.6343907 1.3344650 0.34417635  1.403223 2.285075e-03  3.077466e-03</span></span>
<span><span class="co">#&gt;   adj_p_val_dodr</span></span>
<span><span class="co">#&gt; 1     0.27170004</span></span>
<span><span class="co">#&gt; 2     0.48350547</span></span>
<span><span class="co">#&gt; 3     0.48681880</span></span>
<span><span class="co">#&gt; 4     0.07669958</span></span>
<span><span class="co">#&gt; 5     0.05815103</span></span>
<span><span class="co">#&gt; 6     0.33171207</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">category</span><span class="op">)</span>   <span class="co"># number of genes in the different categories</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; change   gain   loss   same </span></span>
<span><span class="co">#&gt;     39      6     32   1157</span></span></code></pre></div>
<p>In addition to the rhythm parameters (amplitude and phase), there are
boolean columns stating if the feature is rhythmic in NC and HFD and
whether the feature is also differentially rhythmic. These are derived
using tuning parameters <code>rhythm_fdr</code> and
<code>compare_fdr</code> and the adjusted p-values for the two
rhythmicity tests using RAIN and differential rhythmicity test from
DODR.</p>
<p>Recall that the previous two approaches are not specific to
microarray data. Finally, we analyze this data using the linear modeling
framework of limma designed for microarray analysis.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compareRhythms.html">compareRhythms</a></span><span class="op">(</span><span class="va">expr_filtered</span>, exp_design <span class="op">=</span> <span class="va">exp_design</span>, period <span class="op">=</span> <span class="fl">24</span>, </span>
<span>                          method <span class="op">=</span> <span class="st">"limma"</span>, just_classify <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="co"># run with default parameters</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></span>
<span><span class="co">#&gt;                   id category rhythmic_in_NC rhythmic_in_HFD diff_rhythmic</span></span>
<span><span class="co">#&gt; 1 ENSMUSG00000000056     same           TRUE            TRUE         FALSE</span></span>
<span><span class="co">#&gt; 2 ENSMUSG00000000303     same          FALSE            TRUE         FALSE</span></span>
<span><span class="co">#&gt; 3 ENSMUSG00000000555     same           TRUE            TRUE         FALSE</span></span>
<span><span class="co">#&gt; 4 ENSMUSG00000000567     same           TRUE            TRUE         FALSE</span></span>
<span><span class="co">#&gt; 5 ENSMUSG00000000708     same           TRUE           FALSE         FALSE</span></span>
<span><span class="co">#&gt; 6 ENSMUSG00000000876     same           TRUE            TRUE         FALSE</span></span>
<span><span class="co">#&gt;      NC_amp  NC_phase    HFD_amp  HFD_phase adj_p_val_NC_or_HFD adj_p_val_DR</span></span>
<span><span class="co">#&gt; 1 1.0972527 2.3730421 0.75178092 2.18098503        1.906244e-11   0.12104887</span></span>
<span><span class="co">#&gt; 2 0.4474557 5.8966561 0.85843846 0.02256061        2.131371e-02   0.47497910</span></span>
<span><span class="co">#&gt; 3 1.1161226 5.5799337 0.90190671 5.40047547        2.510419e-11   0.34303641</span></span>
<span><span class="co">#&gt; 4 0.8647148 0.2533993 0.62547925 6.26387110        1.970017e-05   0.47532705</span></span>
<span><span class="co">#&gt; 5 0.5127966 2.4419457 0.04522975 2.53218127        2.401673e-03   0.06055589</span></span>
<span><span class="co">#&gt; 6 0.7570742 5.9485326 0.76704298 5.34622495        1.126906e-09   0.05594469</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">category</span><span class="op">)</span>   <span class="co"># number of genes in the different categories</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; change   gain   loss   same </span></span>
<span><span class="co">#&gt;     46      9     33   1025</span></span></code></pre></div>
<p>The returned columns are similar to that from <em>DODR</em> (as they
are both based on the same hypothesis testing approach). The only
difference is that the test for rhythmicity in either group can
performed using a single test, whose adjusted p-value is returned. The
differential rhythmicity test compares circadian parameters (as
coefficients of a harmonic regression) between the two groups. The
boolean columns are constructed (as before) using these p-values.
Moreover, due to the similarity of approaches the results of the
analyses are very similar.</p>
<p>We can also vary the parameters <code>amp_cutoff</code>,
<code>compare_fdr</code> and <code>rhythm_fdr</code> as necessary.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compareRhythms.html">compareRhythms</a></span><span class="op">(</span><span class="va">expr_filtered</span>, exp_design <span class="op">=</span> <span class="va">exp_design</span>, period <span class="op">=</span> <span class="fl">24</span>, </span>
<span>                          method <span class="op">=</span> <span class="st">"limma"</span>, just_classify <span class="op">=</span> <span class="cn">TRUE</span>, amp_cutoff <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">category</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; change   gain   loss   same </span></span>
<span><span class="co">#&gt;     62      1      1   3118</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compareRhythms.html">compareRhythms</a></span><span class="op">(</span><span class="va">expr_filtered</span>, exp_design <span class="op">=</span> <span class="va">exp_design</span>, period <span class="op">=</span> <span class="fl">24</span>, </span>
<span>                          method <span class="op">=</span> <span class="st">"limma"</span>, just_classify <span class="op">=</span> <span class="cn">TRUE</span>, rhythm_fdr <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">category</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; change   gain   loss   same </span></span>
<span><span class="co">#&gt;     41      8     30   1122</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compareRhythms.html">compareRhythms</a></span><span class="op">(</span><span class="va">expr_filtered</span>, exp_design <span class="op">=</span> <span class="va">exp_design</span>, period <span class="op">=</span> <span class="fl">24</span>, </span>
<span>                          method <span class="op">=</span> <span class="st">"limma"</span>, just_classify <span class="op">=</span> <span class="cn">TRUE</span>, compare_fdr <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">category</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; change   gain   loss   same </span></span>
<span><span class="co">#&gt;     97     19     80    917</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="example-2-rna-seq-data">Example 2: RNA-seq data<a class="anchor" aria-label="anchor" href="#example-2-rna-seq-data"></a>
</h3>
<p>We next analyze an RNA-sequencing dataset comparing the effect of a
high fat diet on the mouse liver transcriptome with the same
experimental design <span class="citation">(Quagliarini et al.
2019)</span>. This data too is provided with this package as a 37310x36
matrix called <code>high_fat_diet_rnaseq</code>. We follow a similar
sequence of steps as for microarrays in Example 1.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/bharathananth/compareRhythms" class="external-link">compareRhythms</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioinf.wehi.edu.au/edgeR/" class="external-link">edgeR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/thelovelab/DESeq2" class="external-link">DESeq2</a></span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">high_fat_diet_rnaseq</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;                      HFD_ZT0_1 HFD_ZT0_2 HFD_ZT0_3 HFD_ZT4_1 HFD_ZT4_2</span></span>
<span><span class="co">#&gt; ENSMUSG00000090025.1         0         0         0         0         0</span></span>
<span><span class="co">#&gt; ENSMUSG00000064842.1         0         0         0         0         0</span></span>
<span><span class="co">#&gt; ENSMUSG00000051951.5         0         0         0         0         0</span></span>
<span><span class="co">#&gt; ENSMUSG00000089699.1         0         0         0         0         0</span></span>
<span><span class="co">#&gt; ENSMUSG00000088390.1         0         0         0         0         0</span></span>
<span><span class="co">#&gt; ENSMUSG00000089420.1         0         0         0         0         0</span></span>
<span><span class="co">#&gt;                      HFD_ZT4_3</span></span>
<span><span class="co">#&gt; ENSMUSG00000090025.1         0</span></span>
<span><span class="co">#&gt; ENSMUSG00000064842.1         0</span></span>
<span><span class="co">#&gt; ENSMUSG00000051951.5         2</span></span>
<span><span class="co">#&gt; ENSMUSG00000089699.1         0</span></span>
<span><span class="co">#&gt; ENSMUSG00000088390.1         0</span></span>
<span><span class="co">#&gt; ENSMUSG00000089420.1         0</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">high_fat_diet_rnaseq</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "HFD_ZT0_1"  "HFD_ZT0_2"  "HFD_ZT0_3"  "HFD_ZT4_1"  "HFD_ZT4_2" </span></span>
<span><span class="co">#&gt;  [6] "HFD_ZT4_3"  "HFD_ZT8_1"  "HFD_ZT8_2"  "HFD_ZT8_3"  "HFD_ZT12_1"</span></span>
<span><span class="co">#&gt; [11] "HFD_ZT12_2" "HFD_ZT12_3" "HFD_ZT16_1" "HFD_ZT16_2" "HFD_ZT16_3"</span></span>
<span><span class="co">#&gt; [16] "HFD_ZT20_1" "HFD_ZT20_2" "HFD_ZT20_3" "NC_ZT0_1"   "NC_ZT0_2"  </span></span>
<span><span class="co">#&gt; [21] "NC_ZT0_3"   "NC_ZT4_1"   "NC_ZT4_2"   "NC_ZT4_3"   "NC_ZT8_1"  </span></span>
<span><span class="co">#&gt; [26] "NC_ZT8_2"   "NC_ZT8_3"   "NC_ZT12_1"  "NC_ZT12_2"  "NC_ZT12_3" </span></span>
<span><span class="co">#&gt; [31] "NC_ZT16_1"  "NC_ZT16_2"  "NC_ZT16_3"  "NC_ZT20_1"  "NC_ZT20_2" </span></span>
<span><span class="co">#&gt; [36] "NC_ZT20_3"</span></span></code></pre></div>
<p>We will next construct the data.frame of the experimental design from
the data column names. As before, we fix “NC” to be the reference level
for <em>factor</em> group column.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exp_design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_split.html" class="external-link">str_split</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">high_fat_diet_rnaseq</span><span class="op">)</span>, <span class="st">"_"</span>, simplify <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>     <span class="co"># split the names by _</span></span>
<span><span class="va">exp_design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">exp_design</span>, stringsAsFactors<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>    <span class="co"># convert from matrix to data.frame</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">exp_design</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"group"</span>, <span class="st">"time"</span>, <span class="st">"rep"</span><span class="op">)</span>    <span class="co"># name the columns</span></span>
<span><span class="va">exp_design</span><span class="op">$</span><span class="va">time</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"ZT"</span>, <span class="st">""</span>, <span class="va">exp_design</span><span class="op">$</span><span class="va">time</span><span class="op">)</span><span class="op">)</span> <span class="co"># remove ZT prefix and make numeric</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">exp_design</span><span class="op">)</span></span>
<span><span class="co">#&gt;   group time rep</span></span>
<span><span class="co">#&gt; 1   HFD    0   1</span></span>
<span><span class="co">#&gt; 2   HFD    0   2</span></span>
<span><span class="co">#&gt; 3   HFD    0   3</span></span>
<span><span class="co">#&gt; 4   HFD    4   1</span></span>
<span><span class="co">#&gt; 5   HFD    4   2</span></span>
<span><span class="co">#&gt; 6   HFD    4   3</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">exp_design</span><span class="op">)</span>   <span class="co"># view data type of each column</span></span>
<span><span class="co">#&gt; 'data.frame':    36 obs. of  3 variables:</span></span>
<span><span class="co">#&gt;  $ group: Factor w/ 2 levels "HFD","NC": 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span><span class="co">#&gt;  $ time : num  0 0 0 4 4 4 8 8 8 12 ...</span></span>
<span><span class="co">#&gt;  $ rep  : Factor w/ 3 levels "1","2","3": 1 2 3 1 2 3 1 2 3 1 ...</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Factor-class.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">exp_design</span><span class="op">$</span><span class="va">group</span><span class="op">)</span> <span class="co"># the first level is the reference group</span></span>
<span><span class="co">#&gt; [1] "HFD" "NC"</span></span>
<span><span class="va">exp_design</span><span class="op">$</span><span class="va">group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/relevel.html" class="external-link">relevel</a></span><span class="op">(</span><span class="va">exp_design</span><span class="op">$</span><span class="va">group</span>, <span class="st">"NC"</span><span class="op">)</span> <span class="co"># choose NC as the correct reference group</span></span></code></pre></div>
<p>Next, the data matrix must be filtered down to include only
sufficiently <em>expressed</em> genes. We will used a convenient
function <code><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html" class="external-link">filterByExpr()</a></code> in <a href="https://doi.org/10.18129/B9.bioc.edgeR" class="external-link">edgeR</a> to select genes
that have sufficiently expression (20 counts at the median library size
in 70% of the samples in each group).</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html" class="external-link">filterByExpr</a></span><span class="op">(</span><span class="va">high_fat_diet_rnaseq</span>, group <span class="op">=</span> <span class="va">exp_design</span><span class="op">$</span><span class="va">group</span>, min.count<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">keep</span><span class="op">)</span>   <span class="co"># view the number of genes retained.</span></span>
<span><span class="co">#&gt; keep</span></span>
<span><span class="co">#&gt; FALSE  TRUE </span></span>
<span><span class="co">#&gt; 24375 12935</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">high_fat_diet_rnaseq</span><span class="op">[</span><span class="va">keep</span>, <span class="op">]</span></span></code></pre></div>
<p>At this point, we would like to ensure that there are no outlier
samples. We are using the utilities provided by edgeR but you can use
the steps in the <a href="">DESeq2 vignette</a> instead.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y_explore</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a></span><span class="op">(</span>counts<span class="op">=</span><span class="va">counts</span>, group <span class="op">=</span> <span class="va">exp_design</span><span class="op">$</span><span class="va">group</span><span class="op">)</span></span>
<span><span class="va">y_explore</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html" class="external-link">calcNormFactors</a></span><span class="op">(</span><span class="va">y_explore</span><span class="op">)</span></span>
<span><span class="va">mdsscale</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotMDS.html" class="external-link">plotMDS</a></span><span class="op">(</span><span class="va">y_explore</span>, plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">mdsscale</span><span class="op">$</span><span class="va">x</span>, Y <span class="op">=</span> <span class="va">mdsscale</span><span class="op">$</span><span class="va">y</span>,</span>
<span>                 diet <span class="op">=</span> <span class="va">exp_design</span><span class="op">$</span><span class="va">group</span>,</span>
<span>                 time <span class="op">=</span> <span class="va">exp_design</span><span class="op">$</span><span class="va">time</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">X</span>, <span class="va">Y</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">time</span>, color<span class="op">=</span><span class="va">diet</span><span class="op">)</span>, size<span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span>base_size<span class="op">=</span><span class="fl">10</span><span class="op">)</span> <span class="op">+</span>   <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_color_brewer</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Diet"</span>, palette<span class="op">=</span><span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Leading logFC dim 1"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Leading logFC dim 2"</span><span class="op">)</span></span></code></pre></div>
<p><img src="UserGuide_files/figure-html/MDS-1.png" width="384" style="display: block; margin: auto;"></p>
<p>The samples appear to separate into the two diet groups. A ZT12
sample under NC is likely an outlier.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.min.html" class="external-link">which.max</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">X</span><span class="op">)</span>    <span class="co"># find sample with the largest X deviation (our outlier)</span></span>
<span><span class="va">counts_no_outlier</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span>, <span class="op">-</span><span class="va">ind</span><span class="op">]</span>   <span class="co"># remove that sample</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">counts_no_outlier</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 12935    35</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">exp_design</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">counts_no_outlier</span><span class="op">)</span>   <span class="co">#input-check</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span></code></pre></div>
<p>The experimental design now does not match the count matrix after we
removed the outlier. We have to also remove this sample from the
exp_design.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exp_design_no_outlier</span> <span class="op">&lt;-</span> <span class="va">exp_design</span><span class="op">[</span><span class="op">-</span><span class="va">ind</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">exp_design_no_outlier</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">counts_no_outlier</span><span class="op">)</span>   <span class="co">#input-check</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span></code></pre></div>
<p>Now, the data is ready for <code>compareRhythms</code>. We can use
the <em>voom</em>, <em>DESeq2</em> or <em>edgeR</em>.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compareRhythms.html">compareRhythms</a></span><span class="op">(</span><span class="va">counts_no_outlier</span>, <span class="va">exp_design_no_outlier</span>, method <span class="op">=</span> <span class="st">"voom"</span><span class="op">)</span>  <span class="co"># with default parameters</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">results1</span><span class="op">$</span><span class="va">category</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; change   loss   same </span></span>
<span><span class="co">#&gt;     24      5   2506</span></span>
<span><span class="va">results2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compareRhythms.html">compareRhythms</a></span><span class="op">(</span><span class="va">counts_no_outlier</span>, <span class="va">exp_design_no_outlier</span>, method <span class="op">=</span> <span class="st">"deseq2"</span><span class="op">)</span>  <span class="co"># with default parameters</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">results2</span><span class="op">$</span><span class="va">category</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; change   gain   loss   same </span></span>
<span><span class="co">#&gt;     68     17     16   2789</span></span>
<span><span class="va">results3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compareRhythms.html">compareRhythms</a></span><span class="op">(</span><span class="va">counts_no_outlier</span>, <span class="va">exp_design_no_outlier</span>, method <span class="op">=</span> <span class="st">"edger"</span><span class="op">)</span>  <span class="co"># with default parameters</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">results3</span><span class="op">$</span><span class="va">category</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; change   gain   loss   same </span></span>
<span><span class="co">#&gt;     20      1      4   2642</span></span></code></pre></div>
<p>The <em>voom</em> analysis can take care of outlier samples (without
removing the outlier samples) by setting <code>outliers=TRUE</code>.
This is generally more powerful than removing the outlier sample.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results4</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compareRhythms.html">compareRhythms</a></span><span class="op">(</span><span class="va">counts</span>, <span class="va">exp_design</span>, method <span class="op">=</span> <span class="st">"voom"</span>, outliers <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>  <span class="co"># with default parameters</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">results4</span><span class="op">$</span><span class="va">category</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; change   gain   loss   same </span></span>
<span><span class="co">#&gt;    125     41     32   2943</span></span></code></pre></div>
<p>Of course, these hypothesis-testing based methods can be fine tuned
by changing <code>compare_fdr</code>, <code>rhythm_fdr</code> and
<code>amp_cutoff</code>.</p>
<p>It is also possible to process this data using model selection after
the data are normalized. We use the tools in <em>edgeR</em> to compute
the normalized log counts per million expression of the data and then
apply model selection as in Example 1.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a></span><span class="op">(</span><span class="va">counts_no_outlier</span><span class="op">)</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html" class="external-link">calcNormFactors</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span></span>
<span><span class="va">results5</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compareRhythms.html">compareRhythms</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html" class="external-link">cpm</a></span><span class="op">(</span><span class="va">y</span>, log <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, <span class="va">exp_design_no_outlier</span>, method <span class="op">=</span> <span class="st">"mod_sel"</span><span class="op">)</span>  <span class="co"># with default parameters</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">results5</span><span class="op">$</span><span class="va">category</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  arrhy   loss   gain   same change </span></span>
<span><span class="co">#&gt;   7070    112    295   1519    172</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h3>
<pre><code><span><span class="co">#&gt; R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] DESeq2_1.48.1               SummarizedExperiment_1.38.1</span></span>
<span><span class="co">#&gt;  [3] Biobase_2.68.0              MatrixGenerics_1.20.0      </span></span>
<span><span class="co">#&gt;  [5] matrixStats_1.5.0           GenomicRanges_1.60.0       </span></span>
<span><span class="co">#&gt;  [7] GenomeInfoDb_1.44.0         IRanges_2.42.0             </span></span>
<span><span class="co">#&gt;  [9] S4Vectors_0.46.0            BiocGenerics_0.54.0        </span></span>
<span><span class="co">#&gt; [11] generics_0.1.4              edgeR_4.6.3                </span></span>
<span><span class="co">#&gt; [13] limma_3.64.1                lubridate_1.9.4            </span></span>
<span><span class="co">#&gt; [15] forcats_1.0.0               stringr_1.5.1              </span></span>
<span><span class="co">#&gt; [17] dplyr_1.1.4                 purrr_1.1.0                </span></span>
<span><span class="co">#&gt; [19] readr_2.1.5                 tidyr_1.3.1                </span></span>
<span><span class="co">#&gt; [21] tibble_3.3.0                ggplot2_3.5.2              </span></span>
<span><span class="co">#&gt; [23] tidyverse_2.0.0             compareRhythms_1.1.0       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] tidyselect_1.2.1        DODR_0.99.2             farver_2.1.2           </span></span>
<span><span class="co">#&gt;  [4] fastmap_1.2.0           digest_0.6.37           timechange_0.3.0       </span></span>
<span><span class="co">#&gt;  [7] lifecycle_1.0.4         survival_3.8-3          Rfit_0.27.0            </span></span>
<span><span class="co">#&gt; [10] statmod_1.5.0           magrittr_2.0.3          compiler_4.5.1         </span></span>
<span><span class="co">#&gt; [13] rlang_1.1.6             sass_0.4.10             tools_4.5.1            </span></span>
<span><span class="co">#&gt; [16] yaml_2.3.10             knitr_1.50              S4Arrays_1.8.1         </span></span>
<span><span class="co">#&gt; [19] labeling_0.4.3          DelayedArray_0.34.1     npsm_2.0.0             </span></span>
<span><span class="co">#&gt; [22] plyr_1.8.9              RColorBrewer_1.1-3      BiocParallel_1.42.1    </span></span>
<span><span class="co">#&gt; [25] abind_1.4-8             withr_3.0.2             desc_1.4.3             </span></span>
<span><span class="co">#&gt; [28] grid_4.5.1              multtest_2.64.0         scales_1.4.0           </span></span>
<span><span class="co">#&gt; [31] MASS_7.3-65             cli_3.6.5               rmarkdown_2.29         </span></span>
<span><span class="co">#&gt; [34] crayon_1.5.3            ragg_1.4.0              httr_1.4.7             </span></span>
<span><span class="co">#&gt; [37] tzdb_0.5.0              cachem_1.1.0            splines_4.5.1          </span></span>
<span><span class="co">#&gt; [40] assertthat_0.2.1        parallel_4.5.1          XVector_0.48.0         </span></span>
<span><span class="co">#&gt; [43] vctrs_0.6.5             Matrix_1.7-3            jsonlite_2.0.0         </span></span>
<span><span class="co">#&gt; [46] hms_1.1.3               systemfonts_1.2.3       locfit_1.5-9.12        </span></span>
<span><span class="co">#&gt; [49] jquerylib_0.1.4         glue_1.8.0              pkgdown_2.1.3          </span></span>
<span><span class="co">#&gt; [52] codetools_0.2-20        stringi_1.8.7           gtable_0.3.6           </span></span>
<span><span class="co">#&gt; [55] UCSC.utils_1.4.0        gmp_0.7-5               pillar_1.11.0          </span></span>
<span><span class="co">#&gt; [58] htmltools_0.5.8.1       GenomeInfoDbData_1.2.14 R6_2.6.1               </span></span>
<span><span class="co">#&gt; [61] rain_1.42.0             textshaping_1.0.1       evaluate_1.0.4         </span></span>
<span><span class="co">#&gt; [64] lattice_0.22-7          bslib_0.9.0             class_7.3-23           </span></span>
<span><span class="co">#&gt; [67] Rcpp_1.1.0              SparseArray_1.8.0       xfun_0.52              </span></span>
<span><span class="co">#&gt; [70] fs_1.6.6                pkgconfig_2.0.3</span></span></code></pre>
</div>
<div class="section level3 unnumbered">
<h3 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h3>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-atger_circadian_2015" class="csl-entry">
Atger, Florian, Cédric Gobet, Julien Marquis, Eva Martin, Jingkui Wang,
Benjamin Weger, Grégory Lefebvre, Patrick Descombes, Felix Naef, and
Frédéric Gachon. 2015. <span>“Circadian and Feeding Rhythms
Differentially Affect Rhythmic <span class="nocase">mRNA</span>
Transcription and Translation in Mouse Liver.”</span> <em>Proceedings of
the National Academy of Sciences</em>, November, 201515308. <a href="https://doi.org/10.1073/pnas.1515308112" class="external-link">https://doi.org/10.1073/pnas.1515308112</a>.
</div>
<div id="ref-eckel-mahan_reprogramming_2013" class="csl-entry">
Eckel-Mahan, Kristin L., Vishal R. Patel, Sara de Mateo, Ricardo
Orozco-Solis, Nicholas J. Ceglia, Saurabh Sahar, Sherry A.
Dilag-Penilla, Kenneth A. Dyar, Pierre Baldi, and Paolo Sassone-Corsi.
2013. <span>“Reprogramming of the <span>Circadian</span>
<span>Clock</span> by <span>Nutritional</span>
<span>Challenge</span>.”</span> <em>Cell</em> 155 (7): 1464–78. <a href="https://doi.org/10.1016/j.cell.2013.11.034" class="external-link">https://doi.org/10.1016/j.cell.2013.11.034</a>.
</div>
<div id="ref-love_moderated_2014" class="csl-entry">
Love, Michael I, Wolfgang Huber, and Simon Anders. 2014.
<span>“Moderated Estimation of Fold Change and Dispersion for
<span>RNA</span>-Seq Data with <span>DESeq2</span>.”</span> <em>Genome
Biology</em> 15 (12): 550. <a href="https://doi.org/10.1186/s13059-014-0550-8" class="external-link">https://doi.org/10.1186/s13059-014-0550-8</a>.
</div>
<div id="ref-quagliarini_cistromic_2019" class="csl-entry">
Quagliarini, Fabiana, Ashfaq Ali Mir, Kinga Balazs, Michael Wierer,
Kenneth Allen Dyar, Celine Jouffe, Konstantinos Makris, et al. 2019.
<span>“Cistromic <span>Reprogramming</span> of the <span>Diurnal</span>
<span>Glucocorticoid</span> <span>Hormone</span> <span>Response</span>
by <span>High</span>-<span>Fat</span> <span>Diet</span>.”</span>
<em>Molecular Cell</em> 76 (4): 531–545.e5. <a href="https://doi.org/10.1016/j.molcel.2019.10.007" class="external-link">https://doi.org/10.1016/j.molcel.2019.10.007</a>.
</div>
<div id="ref-ritchie_limma_2015" class="csl-entry">
Ritchie, Matthew E., Belinda Phipson, Di Wu, Yifang Hu, Charity W. Law,
Wei Shi, and Gordon K. Smyth. 2015. <span>“Limma Powers Differential
Expression Analyses for <span>RNA</span>-Sequencing and Microarray
Studies.”</span> <em>Nucleic Acids Research</em> 43 (7): e47–47. <a href="https://doi.org/10.1093/nar/gkv007" class="external-link">https://doi.org/10.1093/nar/gkv007</a>.
</div>
<div id="ref-robinson_edger_2010" class="csl-entry">
Robinson, M. D., D. J. McCarthy, and G. K. Smyth. 2010. <span>“<span class="nocase">edgeR</span>: A <span>Bioconductor</span> Package for
Differential Expression Analysis of Digital Gene Expression
Data.”</span> <em>Bioinformatics</em> 26 (1): 139–40. <a href="https://doi.org/10.1093/bioinformatics/btp616" class="external-link">https://doi.org/10.1093/bioinformatics/btp616</a>.
</div>
<div id="ref-thaben_differential_2016" class="csl-entry">
Thaben, Paul F., and Pål O. Westermark. 2016. <span>“Differential
Rhythmicity: Detecting Altered Rhythmicity in Biological Data.”</span>
<em>Bioinformatics</em> 32 (18): 2800–2808. <a href="https://doi.org/10.1093/bioinformatics/btw309" class="external-link">https://doi.org/10.1093/bioinformatics/btw309</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Bharath Ananthasubramaniam.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
